src/index.ts

```diff
@@ -214,9 +219,14 @@ export default class Kumulos {
         return trackEvent(this.context, type, properties).then(_ => void 0);
     }
 
+    async getPushSubscriptionStatus(subscriptionStateListener: (subscriptionState: PushSubscriptionState) => void) {
```

Comment:
Any particular reason to use a callback here instead of just returning a Promise?

Reply:
The way the methods are currently cached and called requires it to be a listener. The Optimove part exposes it as  as a promise though

Reply:
Ah gotcha because we use the global function invocations & not the class instance directly.